source/mesh_refinement/composition_gradient.cc

```diff
@@ -169,6 +169,10 @@ namespace aspect
                                               "even if the finite element solution is different "
                                               "from each cell to the next. Consequently, the "
                                               "current refinement criterion will likely not be "
-                                              "useful in this situation.")
+                                              "useful in this situation."
+                                              "\n\n"
```

Comment:
any particular reason why you make a paragraph exactly there? I would prefer a new paragraph higher up (before "on the other hand, for discontinuous ...")